src/components/EmbedSource.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react';
+import { Spin } from 'antd';
+import { TwitterTweetEmbed } from 'react-twitter-embed';
+
+const regex = { tweet: /(?<=twitter.com\/(.*)\/status\/)\d*/i };
+
+// This is copy-pasted from src/timeline/TimelineItems.js
+// May make sense to separate into its own file depending on how often we end up needing this functionality
```

Comment:
Good call, I use it in 2 other places. I have a slightly different version I've been using that keeps that .com part but messes up when there's another . in the path, so it's not perfect yet

Reply:
Or maybe we don't like the .com and this is better!

Reply:
I have no opinion either way on including the TLD. My only concern is that currently it doesn't truncate any TLDs other than .com, so twitter.com becomes twitter but wikipedia.org will still be displayed as wikipedia.org.